### PR TITLE
Document the relationship between remacs-sys and remacs

### PR DESCRIPTION
**Pull request by ludwigpacifici**
Here is a first draft which describe the relationship between `remacs-sys` and `remacs`. It is a `README` file, inspired from  `remacs-macros` and `remacs-lib`.

Please let me know what you think. I have the following points in mind:

1. Note that English is not my first language, hence typos or grammar can be wrong. 
2. I aimed a detailed and beginner friendly documentation. Maybe a more tersed documentation would be better?
3. Should I talk about other topic in the `README`?

[ticket: #157]

### CODE REVIEW

File: rust_src/remacs-sys/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# remacs-sys
+
+`remacs-sys` is Rust library containing all [Foreign Function Interface (FFI)](https://doc.rust-lang.org/book/first-edition/ffi.html) declarations of structs and functions that are defined in C. Then, they can be used in the Rust code. C implementations are in `remacs/src`.
```
Comment (Wilfred) [inline rust_src/remacs-sys/README.md]: *is a Rust library containing all the Foreign

File: rust_src/remacs-sys/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+## Example: Use `make_float` C function in Rust
+
+First, let's have a look how `make_float` is defined historically in the C code. Then, how one can re-use it in Rust.
+
+### On the C side
+
+`make_float` is regular C function . The declaration can be found in the header file `remacs/src/lisp.h`:
```
Comment (Wilfred) [inline rust_src/remacs-sys/README.md]: *is a regular C function

I'd be tempted to say "is a C function" -- I'm not sure we gain anything by saying 'regular'.

File: rust_src/remacs-sys/README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Now everything is defined, `make_float` can be used in Rust. For example in `remacs/rust_src/src/lisp.rs`:
+
+```rust
+    #[inline]
+    pub fn from_float(v: EmacsDouble) -> LispObject {
+        LispObject::from_raw(unsafe { make_float(v) })
+    }
+```
+
+Note, foreign functions are considered unsafe so the call is wrapped in a `unsafe` block.
```
Comment (Wilfred) [inline rust_src/remacs-sys/README.md]: *in an unsafe

File: rust_src/remacs-sys/README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+```rust
+#[repr(C)]
+pub struct Lisp_Hash_Table {
+    pub header: Lisp_Vectorlike_Header,
+    // ...
+}
+```
+
+All types of `Lisp_Hash_Table` have to be defined as well. Then, under the hood, C and Rust use different data layout. [`repr(C)`](https://doc.rust-lang.org/nomicon/other-reprs.html#reprc) is used to ensure the compatibility between the two.
```
Comment (Wilfred) [inline rust_src/remacs-sys/README.md]: *different data lyouts

File: rust_src/remacs-sys/README.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+Finally, `Lisp_Hash_Table` struct can be used in Rust and abso be passed to C functions. For example, in `remacs/rust_src/src/hashtable.rs`:
```
Comment (Wilfred) [inline rust_src/remacs-sys/README.md]: *the Lisp_Hash_Table struct

*and also

**Comment by ludwigpacifici** [comment]
Thanks @Wilfred for your feedback :-)

I agree with your suggestions. it's committed. Let me know if other changes are required.

**Pull request merged**